packages/dataviews/src/view-list.tsx

```diff
@@ -142,23 +205,136 @@ function ListItem( {
 						</HStack>
 					</CompositeItem>
 				</div>
+				{ actions?.length > 0 && (
+					<HStack
+						spacing={ 1 }
+						justify="flex-end"
+						className="dataviews-view-list__item-actions"
+						style={ {
+							flexShrink: '0',
+							width: 'auto',
+						} }
+					>
+						{ primaryAction && !! primaryAction.RenderModal && (
+							<div role="gridcell">
+								<CompositeItem
+									store={ store }
+									render={
+										<Button
+											label={ primaryAction.label }
+											icon={ primaryAction.icon }
+											isDestructive={
+												primaryAction.isDestructive
+											}
+											size="compact"
+											onClick={ () =>
+												setIsModalOpen( true )
+											}
+										/>
+									}
+								>
+									{ isModalOpen && (
+										<ActionModal
+											action={ primaryAction }
+											items={ [ item ] }
+											closeModal={ () =>
+												setIsModalOpen( false )
+											}
+											onActionStart={ () => {} }
+											onActionPerformed={ () => {} }
```

Comment:
Why pass empty functions at all?

Reply:
Typescript :)

Thanks for bring it up. I added this at some point so Typescript let me focus on behavioral issues rather than ortographical things but forgot to add a `TODO` marker to come back to it later. Prepared a follow-up to fix this at https://github.com/WordPress/gutenberg/pull/61659